mslib/utils/mssautoplot.py

```diff
@@ -77,25 +85,98 @@ def load_from_ftml(filename):
+    if verify_user_token(mscolab_server_url, token):
+        data = {
+            "token": token,
+            "op_id": op_id
+        }
+        url = urljoin(mscolab_server_url, "get_operation_by_id")
+        r = requests.get(url, data=data)
+        if r.text != "False":
+            xml_content = json.loads(r.text)["content"]
+            return xml_content
+
+
+def get_op_id(token, mscolab_server_url, curr_op):
+    logging.debug('get_recent_op_id')
+    if verify_user_token(mscolab_server_url, token):
+        """
+        get most recent operation's op_id
+        """
+        skip_archived = config_loader(dataset="MSCOLAB_skip_archived_operations")
+        data = {
+            "token": token,
+            "skip_archived": skip_archived
+        }
+        url = urljoin(mscolab_server_url, "operations")
+        r = requests.get(url, data=data)
+        if r.text != "False":
+            _json = json.loads(r.text)
+            operations = _json["operations"]
+            for op in operations:
+                if op["path"] == curr_op:
+                    return op["op_id"]
+
+
 class Plotting:
-    def __init__(self, cpath):
+    def __init__(self, cpath, username=None, password=None, mscolab_server_url=None):
         read_config_file(cpath)
         self.config = config_loader()
         self.num_interpolation_points = self.config["num_interpolation_points"]
         self.num_labels = self.config["num_labels"]
         self.tick_index_step = self.num_interpolation_points // self.num_labels
         self.bbox = None
+        flight = self.config["automated_plotting_flights"][0][0]
         section = self.config["automated_plotting_flights"][0][1]
         filename = self.config["automated_plotting_flights"][0][3]
-        try:
-            self.params = mslib.utils.coordinate.get_projection_params(
-                self.config["predefined_map_sections"][section]["CRS"].lower())
-        except KeyError as e:
-            print(e)
-            sys.exit("Invalid SECTION and/or CRS")
-        self.params["basemap"].update(self.config["predefined_map_sections"][section]["map"])
-        self.bbox_units = self.params["bbox"]
-        self.read_ftml(filename)
+        if self.__class__.__name__ == "TopViewPlotting":
+            try:
+                self.params = mslib.utils.coordinate.get_projection_params(
+                    self.config["predefined_map_sections"][section]["CRS"].lower())
+            except KeyError as e:
+                print(e)
+                sys.exit("Invalid SECTION and/or CRS")
+            self.params["basemap"].update(self.config["predefined_map_sections"][section]["map"])
+            self.bbox_units = self.params["bbox"]
+        if filename != "" and filename == flight:
+            self.read_operation(flight, username, password, mscolab_server_url)
+        elif filename != "":
+            self.read_ftml(filename)
 
     def read_ftml(self, filename):
         dirpath = "./"
```

Comment:
ToDo: default should be the dir MSUI_SETTINGS dir where msui stores the file. The current path should be optional.

This blocks its usage from the autoplot dockingwidget, when the data file is not found.

Reply:
the cli should have a parameter to bypass directories defined in the mssautoplot.json for ftml file, and on default we should store fullpath of the file.